pandas/tests/test_reshape.py

```diff
@@ -411,6 +411,91 @@ def test_dataframe_dummies_with_categorical(self):
                              ]]
         assert_frame_equal(result, expected)
 
+    # GH12402 Add a new parameter `drop_first` to avoid collinearity
+    def test_basic_drop_first(self):
+        # Basic case
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_series_index = Series(s_list, list('ABC'))
+
+        expected = DataFrame({'b': {0: 0.0,
+                                    1: 1.0,
+                                    2: 0.0},
+                              'c': {0: 0.0,
+                                    1: 0.0,
+                                    2: 1.0}})
+
+        result = get_dummies(s_list, sparse=self.sparse, drop_first=True)
+        assert_frame_equal(result, expected)
+
+        result = get_dummies(s_series, sparse=self.sparse, drop_first=True)
+        assert_frame_equal(result, expected)
+
```

Comment:
@jreback By "empty frame", do you mean this case
```python
In [1]: import pandas as pd

In [2]: pd.get_dummies(pd.Series())
Out[2]:
Empty DataFrame
Columns: []
Index: []

In [3]: pd.get_dummies(pd.Series(), drop_first=True)
Out[3]:
Empty DataFrame
Columns: []
Index: []
```
Or this case
```python
In [4]: pd.get_dummies(pd.DataFrame())

```

But this will raise an error
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-059be6e25969> in <module>()
----> 1 pd.get_dummies(pd.DataFrame())

C:\D\Projects\Github\pandas\pandas\core\reshape.py in get_dummies(data, prefix,
prefix_sep, dummy_na, columns, sparse, drop_first)
   1083                                     drop_first=drop_first)
   1084             with_dummies.append(dummy)
-> 1085         result = concat(with_dummies, axis=1)
   1086     else:
   1087         result = _get_dummies_1d(data, prefix, prefix_sep, dummy_na,

C:\D\Projects\Github\pandas\pandas\tools\merge.py in concat(objs, axis, join, jo
in_axes, ignore_index, keys, levels, names, verify_integrity, copy)
    833                        keys=keys, levels=levels, names=names,
    834                        verify_integrity=verify_integrity,
--> 835                        copy=copy)
    836     return op.get_result()
    837

C:\D\Projects\Github\pandas\pandas\tools\merge.py in __init__(self, objs, axis,
join, join_axes, keys, levels, names, ignore_index, verify_integrity, copy)
    866
    867         if len(objs) == 0:
--> 868             raise ValueError('No objects to concatenate')
    869
    870         if keys is None:

ValueError: No objects to concatenate
```